python3Packages.pyobjc: Reinit

SuperSandro2000 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

##### ToDo

- [x] Verify update script
- [x] Re-run update script and check for missing modules
- [x] reflow imports
- [x] force push all of the above
- [ ] ~~nixpkgs-review~~ rebuilds everything so not really possible

##### Python packages to unbreak after this is merged:

- https://github.com/NixOS/nixpkgs/pull/102643#issuecomment-721455344

###### Motivation for this change

#21829
Closes #101360

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 committed:
python3Packages.pyobjc: reinit at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-core: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.GameKit: Fix Headers/GameKit.h:9:9: fatal error: 'simd/simd.h' file not found

SuperSandro2000 committed:
apple-sdk: sort impure deps alphabetically

SuperSandro2000 committed:
apple_sdk.frameworks.WebKit: Fix WebKit.framework/Headers/WKNavigationAction.h:33:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
apple_sdk.frameworks.SceneKit: Fix Headers/SceneKitTypes.h:12:9: fatal error: 'GLKit/GLKMathTypes.h' file not found

SuperSandro2000 committed:
apple_sdk.frameworks.CoreVideo: Fix Headers/CVMetalTextureCache.h:42:9: fatal error: 'Metal/MTLPixelFormat.h' file not found

SuperSandro2000 committed:
apple_sdk.frameworks.Quartz: Fix Headers/QCView.h:5:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
apple_sdk.frameworks.Quartz: Fix Headers/QuartzFilter.h:10:9: fatal error: 'Cocoa/Cocoa.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Cocoa: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Collaboration: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreServices: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-FSEvents: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-GameKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MetalKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Quartz: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SearchKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ServiceManagement: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SystemExtensions: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-VideoToolbox: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-WebKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreMedia: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ColorSync: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-FinderSync: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-NetFS: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SyncServices: ini at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreData: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.MapKit: Fix Headers/MKAnnotation.h:9:9: fatal error: 'CoreLocation/CoreLocation.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MapKit: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.MapKit: Fix Headers/MKAnnotationView.h:12:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreLocation: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Accounts: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CloudKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-DictionaryServices: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-libdispatch: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add NotificationCenter

SuperSandro2000 committed:
python3Packages.pyobjc-framework-NotificationCenter: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Speech: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AddressBook: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Metal: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.AVKit: Fix Headers/AVCaptureView.h:10:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AVKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-DiscRecording: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-DiscRecordingUI: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add NetworkExtension

SuperSandro2000 committed:
python3Packages.pyobjc-framework-NetworkExtension: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-GameCenter: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MediaToolbox: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Security: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SecurityFoundation: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-IOSurface: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-PencilKit: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.MetalKit: Fix Headers/MTKView.h:11:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ExceptionHandling: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AVFoundation: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-DVDPlayback: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CalendarStore: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add SafariServices

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SafariServices: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AppleScriptKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AdSupport: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-StoreKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreText: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ImageCaptureCore: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ApplicationServices: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-LocalAuthentication: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreMediaIO: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add Intents

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Intents: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SystemConfiguration: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.SecurityInterface: Fix Headers/SFAuthorizationView.h:11:9: fatal error: 'Cocoa/Cocoa.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SecurityInterface: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ExecutionPolicy: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-InputMethodKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreHaptics: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-GameController: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-VideoSubscriberAccount: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreBluetooth: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.ScreenSaver: Fix Headers/ScreenSaverView.h:8:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ScreenSaver: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-OSAKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Photos: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add Photo

SuperSandro2000 committed:
python3Packages.pyobjc-framework-iTunesLibrary: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-LatentSemanticMapping: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.CoreAudioKit: Fix Headers/AUGenericView.h:14:9: fatal error: 'Cocoa/Cocoa.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreAudioKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreAudio: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Network: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreMotion: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-QuickLookThumbnailing: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-InstantMessage: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Automator: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-DeviceCheck: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreWLAN: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.SpriteKit: Fix Headers/SpriteKitBase.h:18:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
apple_sdk: Sort alphabetically

SuperSandro2000 committed:
apple_sdk.frameworks.SpriteKit: Fix Headers/SKWarpGeometry.h:12:9: fatal error: 'simd/simd.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-NaturalLanguage: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ScriptingBridge: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SpriteKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AuthenticationServices: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CryptoTokenKit: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add CryptoTokenKit

SuperSandro2000 committed:
apple_sdk.frameworks: Add ContactsUI

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ContactsUI: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add Contacts

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Contacts: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-PreferencePanes: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-OpenDirectory: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-UserNotifications: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CFNetwork: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ExternalAccessory: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-GameplayKit: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.GameplayKit: Fix Headers/GameplayKitBase.h:9:9: fatal error: 'simd/simd.h' file not found

SuperSandro2000 committed:
apple_sdk.frameworks.GameplayKit: Fix Headers/GKNoise.h:9:9: fatal error: 'SpriteKit/SpriteKitBase.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-PushKit: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.ModelIO: Fix Headers/MDLTypes.h:13:10: fatal error: 'simd/simd.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-ModelIO: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AppleScriptObjC: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MediaLibrary: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-DiskArbitration: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SoundAnalysis: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MediaAccessibility: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-SceneKit: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.SceneKit: Fix Headers/SceneKitTypes.h:9:9: fatal error: 'simd/simd.h' file not found

SuperSandro2000 committed:
apple_sdk.frameworks.SceneKit: Fix Headers/SCNView.h:7:9: fatal error: 'AppKit/AppKit.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-IMServicePlugIn: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks.IMServicePlugIN: Fix ld: framework not found GameKit

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MediaPlayer: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-OSLog: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-LaunchServices: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreML: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-FileProvider: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-EventKit: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AutomaticAssessmentConfiguration: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Vision: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-LinkPresentation: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-MultipeerConnectivity: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-InstallerPlugins: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-FileProviderUI: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-Social: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CoreSpotlight: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc-framework-BusinessChat: init at 6.2.2

SuperSandro2000 committed:
python3Packages.pyobjc: Add update script

SuperSandro2000 committed:
python3Packages.pyobjc-framework-PhotosUI: init at 6.2.2

SuperSandro2000 committed:
apple_sdk.frameworks: Add PhotoUI

Mic92 commented:
Maybe also @dasJ

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-CalendarStore/default.nix:
```diff
@@ -0,0 +1,49 @@
+      --replace 'os.path.basename(data)[6:-4]' '"${darwin.apple_sdk.sdk.version}"'
+  '';
+
+  buildInputs = [
+  ] ++ (with darwin; [
+  ] ++ (with apple_sdk.frameworks;[
+  ]));
```
Should those empty blocks stay there all the time?

SuperSandro2000 replied:
I can try to remove them and I think I never use the first one anyway.

Mic92 commented:
> How are these packages going to be maintained?

They are largely auto-generated as you can see from the script at the end.

FRidh commented:
How are these packages going to be maintained?

SuperSandro2000 commented:
> How are these packages going to be maintained?

The update script should have a version in the top and if a new version is released you change that, run it and fix any problems or add new modules.

Mic92 commented:
I cherry-picked the mass-rebuilding change here: https://github.com/NixOS/nixpkgs/pull/102348

dasJ commented:
@Mic92 feel free to ping me again when Hydra did all the mass-rebuild stuff

SuperSandro2000 committed:
python3Packages.pyobjc-frameworks-*: Remove empty buildInputs blocks

SuperSandro2000 committed:
python3Packages.pyobjc-frameworks-*: Remove setuptoolsCheckFlagsArray to avoid future deprecation and mass rebuild

SuperSandro2000 committed:
python3Packages.pyobjc-frameworks-*: Fix whitespace issue, remove empty buildInputs

SuperSandro2000 commented:
Is it rebuilding everything darwin because of ``pkgs/os-specific/darwin/apple-sdk/frameworks.nix`` ?

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-Photos/default.nix:
```diff
@@ -11,7 +11,7 @@ buildPythonPackage rec {
     inherit pname version;
     sha256 = "0xgj5h22cijihir97dbwbblkxn55hx5xdicmiwfn8a1lky7mklvd";
   };
-  
```
Minor nitpick, but our editorconfig check, we have in github action, usually does not like trailing whitespace. It would be possible to exclude those files so.

SuperSandro2000 replied:
No longer required. I had some whitespace before the ``<% if...`` in the template file and they where written to the final file.

veprbl commented on pkgs/top-level/python-packages.nix:
```diff
@@ -98,12 +98,15 @@ let
     if x then throw "${removePythonPrefix (drv.pname or drv.name)} not supported for interpreter ${python.executable}" else drv;
 
+  enabledIfDarwin = drv:
+    if (!stdenv.isDarwin) then throw "${removePythonPrefix (drv.pname or drv.name)} can only be built on Darwin" else drv;
```
Let's not add more premature throws. The mechanism behind `meta.platforms` is perfectly capable of displaying the error message without messing with introspection too much.

SuperSandro2000 replied:
Yeah you are right. I think I copied this from the very old PR and did not think to much if it is actually required.

SuperSandro2000 committed:
python-packages: Remove enabledIfDarwin

dasJ commented:
Progress on Darwin so far: `[21/27872 built (5 failed), 1/15439/24483 copied (30937.3/100120.9 MiB), 15716.3/38630.9 MiB DL]`

dasJ commented:
The last failed build I see is:

```
  ld: warning: passed two min versions (10.12.0, 10.12) for platform macOS. Using 10.12.
  ld: library not found for -lunistring
  clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [Makefile:624: doboot.shared] Error 1
  make[2]: Leaving directory '/private/tmp/nix-build-bigloo-4.3h.drv-0/bigloo-4.3h/comptime'
  make[1]: *** [Makefile:616: boot] Error 2
  make[1]: Leaving directory '/private/tmp/nix-build-bigloo-4.3h.drv-0/bigloo-4.3h/comptime'
  make: *** [Makefile:169: boot-c] Error 2
builder for '/nix/store/ji10s1mk80ldcd7mc0ia8dbyba12p604-perl5.30.3-Mac-Pasteboard-0.009.drv' failed with exit code 2; last 10 log lines:

  Test Summary Report
  -------------------
  t/basic.t     (Wstat: 256 Tests: 11 Failed: 1)
    Failed test:  11
    Non-zero exit status: 1
  Files=2, Tests=16,  1 wallclock secs ( 0.01 usr  0.01 sys +  0.12 cusr  0.20 csys =  0.34 CPU)
  Result: FAIL
  Failed 1/2 test programs. 1/16 subtests failed.
  make: *** [Makefile:1100: test_dynamic] Error 255
builder for '/nix/store/v558afp5759wh9ixz6bfy3nkn2mcdgr5-perl5.32.0-Mac-Pasteboard-0.009.drv' failed with exit code 2; last 10 log lines:

  Test Summary Report
  -------------------
  t/basic.t     (Wstat: 256 Tests: 11 Failed: 1)
    Failed test:  11
    Non-zero exit status: 1
  Files=2, Tests=16,  0 wallclock secs ( 0.01 usr  0.01 sys +  0.12 cusr  0.19 csys =  0.33 CPU)
  Result: FAIL
  Failed 1/2 test programs. 1/16 subtests failed.
  make: *** [Makefile:1101: test_dynamic] Error 255
cannot build derivation '/nix/store/4fnnc67hv23j9q42gai7b2vrfq25bwr7-kpcli-3.4.drv': 1 dependencies couldn't be built

```

dasJ commented:
Also
```
builder for '/nix/store/84acgp778yyynrsaphic1bwvmv3bav6k-apk-tools-2.10.5.drv' failed with exit code 2; last 10 log lines:
  make[1]: *** [Make.rules:202: libfetch/fetch.o] Error 127
  /nix/store/8i9kyqfxhk2s9dzx32x0sgdk7kyx9wjc-bash-4.4-p23/bin/bash: gcc: command not found
  /nix/store/8i9kyqfxhk2s9dzx32x0sgdk7kyx9wjc-bash-4.4-p23/bin/bash: gcc: command not found
  /nix/store/8i9kyqfxhk2s9dzx32x0sgdk7kyx9wjc-bash-4.4-p23/bin/bash: gcc: command not found
  /nix/store/8i9kyqfxhk2s9dzx32x0sgdk7kyx9wjc-bash-4.4-p23/bin/bash: gcc: command not found
  make[1]: *** [Make.rules:202: libfetch/file.o] Error 127
  make[1]: *** [Make.rules:202: libfetch/http.o] Error 127
  make[1]: *** [Make.rules:202: libfetch/openssl-compat.o] Error 127
  make[1]: *** [Make.rules:202: libfetch/ftp.o] Error 127
  make: *** [Make.rules:288: libfetch/] Error 2
```

dasJ commented:
```
builder for '/nix/store/mxisz0gk3b21wi3wdp406w9vv6kjs5h2-python2.7-ECPy-1.2.3.drv' failed with exit code 1; last 10 log lines:
  configuring
  no configure script, doing nothing
  building
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 22, in <module>
      reqs.append('future')
  NameError: name 'reqs' is not defined
```

dasJ commented:
```
builder for '/nix/store/rfn6q5xvn2blk4qnaszxa5by133d0p7r-python2.7-pyelftools-unstable-2020-09-23.drv' failed with exit code 127; last 10 log lines:
  post-installation fixup
  strip is /nix/store/1h9ivag16gyysxjgfwq5dfk8w6g246g7-cctools-binutils-darwin-949.0.1/bin/strip
  stripping (with command strip and flags -S) in /nix/store/rhdah2asvp24cmp49p34jhmzgkvaxrsy-python2.7-pyelftools-unstable-2020-09-23/lib  /nix/store/rhdah2asvp24cmp49p34jhmzgkvaxrsy-python2.7-pyelftools-unstable-2020-09-23/bin
  patching script interpreter paths in /nix/store/rhdah2asvp24cmp49p34jhmzgkvaxrsy-python2.7-pyelftools-unstable-2020-09-23
  Rewriting #!/nix/store/ys6xv41f2lbi4mqnrp1b5v326545f6qc-python-2.7.18/bin/python2.7 to #!/nix/store/ys6xv41f2lbi4mqnrp1b5v326545f6qc-python-2.7.18
  wrapping `/nix/store/rhdah2asvp24cmp49p34jhmzgkvaxrsy-python2.7-pyelftools-unstable-2020-09-23/bin/readelf.py'...
  Executing pythonRemoveTestsDir
  Finished executing pythonRemoveTestsDir
  running install tests
  /nix/store/d61m18vxxwqshv4vik4jw3i7a9d3sflx-stdenv-darwin/setup: line 1309: patchelf: command not found
```

```
builder for '/nix/store/0n1nq502fdpxyaz1hqphz8vapx1cxk6y-python2.7-python-can-3.3.4.drv' failed with exit code 1; last 10 log lines:
      PytestUnknownMarkWarning,

  test/test_viewer.py::CanViewerTest::test_pack_unpack
  test/test_viewer.py::CanViewerTest::test_pack_unpack
  test/test_viewer.py::CanViewerTest::test_pack_unpack
    /private/tmp/nix-build-python2.7-python-can-3.3.4.drv-0/python-can-3.3.4/test/test_viewer.py:263: DeprecationWarning: integer argument expected, got float
      return struct_t.pack(*data)

  -- Docs: https://docs.pytest.org/en/latest/warnings.html
  ======== 1 failed, 151 passed, 21 skipped, 4 warnings in 23.25 seconds =========
```

```
builder for '/nix/store/acdvpx4lm78iz2yjk5wdyrgmss0vcvdj-perl5.32.0-Safe-Hole-0.14.drv' failed with exit code 255; last 10 log lines:
  no configure script, doing nothing
  building
  ld: unknown option: -mmacosx-version-min=10.12
  Warning: ExtUtils::CBuilder not installed or no compiler detected
  Proceeding with configuration, but compilation may fail during Build

  Created MYMETA.yml and MYMETA.json
  Creating new 'Build' script for 'Safe-Hole' version '0.14'
  Building Safe-Hole
  Error: no compiler detected to compile 'lib/Safe/Hole.c'.  Aborting
cannot build derivation '/nix/store/i231jgcc5ny0na46i2v54qxdni14faq4-perl5.32.0-String-Interpolate-0.32.drv': 1 dependencies couldn't be built
```

dasJ commented:
```
builder for '/nix/store/j9cg83qiydi3j82dqb862k06ql8fjf6h-watchman-4.9.0.drv' failed with exit code 2; last 10 log lines:
  watcher/fsevents.cpp:499:5: error: no matching function for call to 'CFRelease'
      CFRelease(fdref);
      ^~~~~~~~~
  /nix/store/cx86fgsqp9dkx2h88rq7q73hhh0an9fy-swift-corefoundation/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:654:6: note: candidate function not viable: no known conversion from 'watchman::FileDescriptor' to 'CFTypeRef'
(aka 'const void *') for 1st argument; take the address of the argument with &
  void CFRelease(CFTypeRef cf);
       ^
  7 errors generated.
  make[1]: *** [Makefile:4099: watcher/watchman-fsevents.o] Error 1
  make[1]: Leaving directory '/private/tmp/nix-build-watchman-4.9.0.drv-0/source'
  make: *** [Makefile:1263: all] Error 2
```

veprbl commented:
If this is a mass rebuild, it will have to go to the staging branch.

FRidh commented on pkgs/development/python-modules/pyobjc/default.nix.esh:
```diff
@@ -0,0 +1,108 @@
+  '';
+<% else -%>
+  dontUseSetuptoolsCheck = true;
+  pythonImportcheck = [ "pyobjc-framework-<%= $framework %>" ];
```
`pythonImportCheck` with Capital C

mweinelt replied:
`pythonImportsCheck` with an s.

SuperSandro2000 commented:
@dasJ I think you are listing every package that does not build on darwin right now. I don't think my changes can break anything in darwin other than python things depending on pyobjc. The addition of new framework dependencies should just work. ™️ 

@veprbl Can I just rebase on staging and change the target branch or what is the correct way to do this?

SuperSandro2000 committed:
python3Packages.pyobjc-framework-*: Fix pythonImportsCheck

dasJ commented:
@SuperSandro2000 idk, it's what nixpkgs-review did for `nixpkgs-review pr --post-result 102306`

SuperSandro2000 commented:
> @SuperSandro2000 idk, it's what nixpkgs-review did for `nixpkgs-review pr --post-result 102306`

It is basically doing what ofborg labeled here plus minus some. It is rebuilding everything because I needed to change some frameworks which seem to be widely used. Unfortunately I did not yet figure out how to tell nixpkgs-review to only build pyobjc but you can just checkout the branch and build python3Packages.pyobjc which also builds all other modules.

dasJ commented:
Result of `nixpkgs-review pr 102306` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python3.pkgs.pyobjc</li>
  </ul>
</details>

dasJ commented:
Well do I need to install quicktime for this?

veprbl commented:
> Can I just rebase on staging and change the target branch or what is the correct way to do this?

Yes, this is how we usually do it. It may cause some unnecessary pings when you push staging-based branch into what is a PR targeting master. To my best understanding, there isn't a reliable solution to prevent that from happening.

SuperSandro2000 commented:
> QuickTime

Which package pulls this in? I did not require to install it but I think it is installed.

SuperSandro2000 commented:
Rebased the PR on staging.

veprbl commented:
@SuperSandro2000 Would it make sense to have a separate PR for changes to `apple_sdk.frameworks`? I suppose, we would want some of the authors of that code to look at your changes, but they might not be as interested in the new python packages. Also, I have a question: does this add frameworks only available beyond the 10.12 SDK?

veprbl commented on pkgs/os-specific/darwin/apple-sdk/frameworks.nix:
```diff
@@ -107,7 +107,7 @@ with frameworks; with libs; {
   Tcl                     = [];
   VideoDecodeAcceleration = [ CoreVideo ];
   VideoToolbox            = [ CoreMedia CoreVideo ];
```
Notice typo in the commit message `s/applw/apple/`

SuperSandro2000 replied:
Fixed

SuperSandro2000 commented:
> @SuperSandro2000 Would it make sense to have a separate PR for changes to `apple_sdk.frameworks`?

To be honest not really. I think those compilation errors only occur if some code paths are used which happened to be when I tried to compile the native code in the python modules.

> I suppose, we would want some of the authors of that code to look at your changes, but they might not be as interested in the new python packages.

It does not make sense if no packages are compiled against those files. Without the code in the python modules the errors I encountered don't happen. I could move those includes into the python modules which is probably not a great idea because the errors don't happen in the python package but in the frameworks. Probably many more are incomplete and are missing dependencies but this will only be noticed if we compile more programs against them.

> Also, I have a question: does this add frameworks only available beyond the 10.12 SDK?

Yes. I added comments to them to clarify that. I didn't want to exclude those python modules and start adding them when switching to apple sdk 13, 14 or 15. I think this also makes adding anything compiled against those modules easier because you can look them up that they are currently not included and mark a package broken until we use that SDK version. It also makes future PRs which switch to newer sdk versions (#101940) a little bit smaller.

Maybe we should move them to the end of the file or throw an error if you try to use them. I am not sure which is the best solution here.

Also some frameworks I added where already included in 10.12 but never used by anything and so not included in the list. This makes me think that the list is incomplete even after my additions and maybe will always be incomplete.

jonringer commented:
seeing as this largely only builds for darwin, it's going to be on darwin contributors to keep this up-to-date.  and the current python-unstable workflow doesn't build darwin packages.

Just giving a heads up that @FRidh and I aren't going to be aware of breakages, and will likely not contribute much either.

Mic92 commented:
It seems that this package is also a good test for `apple_sdk` as it make sure everything is specified properly and can be linked/included.

SuperSandro2000 commented:
> seeing as this largely only builds for darwin, it's going to be on darwin contributors to keep this up-to-date.

This PR is basically darwin only.

> and the current python-unstable workflow doesn't build darwin packages.
> I aren't going to be aware of breakages, and will likely not contribute much either.

I have planned to maintain this at least in the near future. If someone from darwin wants to jump on and help me feel free to raise your voice.

> It seems that this package is also a good test for apple_sdk as it make sure everything is specified properly and can be linked/included.

You are right. It does not test everything possible but it is a good starting point to find low hanging bugs.

veprbl commented:
> > @SuperSandro2000 Would it make sense to have a separate PR for changes to `apple_sdk.frameworks`?
> 
> To be honest not really. I think those compilation errors only occur if some code paths are used which happened to be when I tried to compile the native code in the python modules.

The reason why ask is because right now this PR is very big. While it looks good already, I think it may take some time before someone puts in work to review all of it. Do you think it will be possible to factorize the expressions for the python packages using the power of Nix. Right now they have a lot of repeating parts. I think, if the packages were expressed in a more concise way, they would be easier to review/maintain (I do note that you provide an update script, that helps a lot too).

> Also some frameworks I added where already included in 10.12 but never used by anything and so not included in the list. This makes me think that the list is incomplete even after my additions and maybe will always be incomplete.

I think we do want to at least add the missing 10.12 stuff.

SuperSandro2000 committed:
python3Packages.pyobjc-framework-AutomaticAssessmentConfiguration: Mark OSX 10.15 only

SuperSandro2000 committed:
python3Packages.pyobjc-framework-*: Fix OSX check logic

Mic92 commented:
> > > @SuperSandro2000 Would it make sense to have a separate PR for changes to `apple_sdk.frameworks`?
> > 
> > 
> > To be honest not really. I think those compilation errors only occur if some code paths are used which happened to be when I tried to compile the native code in the python modules.
> 
> The reason why ask is because right now this PR is very big. While it looks good already, I think it may take some time before someone puts in work to review all of it. Do you think it will be possible to factorize the expressions for the python packages using the power of Nix. Right now they have a lot of repeating parts. I think, if the packages were expressed in a more concise way, they would be easier to review/maintain (I do note that you provide an update script, that helps a lot too).
> 
> > Also some frameworks I added where already included in 10.12 but never used by anything and so not included in the list. This makes me think that the list is incomplete even after my additions and maybe will always be incomplete.
> 
> I think we do want to at least add the missing 10.12 stuff.

I don't see that many redundant parts. Patching `darwin.apple_sdk.sdk.version` into `pyobjc_setup.py` could be moved into a setup hook. Maybe also the files should say that they are auto-generated so people are not starting to modify them directly.

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-AVFoundation/default.nix:
```diff
@@ -0,0 +1,58 @@
+    # testConstants in PyObjCTest.test_cfsocket.TestSocket returns: Segmentation fault: 11
+    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
+    # Remove Test which is probably missing a sdk check
+    substituteInPlace PyObjCTest/test_avcapturedevice.py \
+      --replace 'def testMissingConstants(self):' 'def disabled_testMissingConstants(self):'
```
With `pytestCheckHook` you can use `disabledTests = [ "testMissingConstants" ];`

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-AddressBook/default.nix:
```diff
@@ -0,0 +1,64 @@
+  preCheck = ''
+    # testConstants in PyObjCTest.test_cfsocket.TestSocket returns: Segmentation fault: 11
+    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
+    # Test removed because it requires network:
+    # an error occurred while attempting to obtain endpoint for listener 'HostCallsAuxiliary': Connection interrupted
+    rm PyObjCTest/test_abpersonview.py
+
+    # Test removed because it likely requires network:
+    # AssertionError: None is not an instance of <objective-c class ABAddressBook at 0x7fff96826300>
+    rm PyObjCTest/test_abaddressbookc.py
```
Also use `pytestCheckHook` here.

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-CFNetwork/default.nix:
```diff
@@ -0,0 +1,54 @@
+  preCheck = ''
+    # testConstants in PyObjCTest.test_cfsocket.TestSocket returns: Segmentation fault: 11
+    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
+    # Remove Test which requires network
+    rm PyObjCTest/test_cfhttpstream.py PyObjCTest/test_cfproxysupport.py PyObjCTest/test_cfhttpmessage.py
```
`pytestCheckHook`

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-Cocoa/tests.patch:
```diff
@@ -0,0 +1,166 @@
+
+-    @min_os_level("10.8")
+-    @expectedFailure
+-    def testConstants10_8(self):
```
Also here I would rather ignore the tests with pytestCheckHook as a whole rather than patching, because patching causes too much maintenance overhead.

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-Metal/default.nix:
```diff
@@ -0,0 +1,64 @@
+    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
+    # Remove Test which is probably missing a sdk check
+    # AttributeError: MTLSamplePositionMake
+    substituteInPlace PyObjCTest/test_mtltypes.py \
+      --replace 'v = Metal.MTLSamplePositionMake(0.5, 1.5)' "" \
+      --replace 'self.assertIsInstance(v, Metal.MTLSamplePosition)' "" \
+      --replace 'self.assertEqual(v, (0.5, 1.5))' ""
+
+    # Remove Test which is probably missing a sdk check
+    # AttributeError: MTLIndirectCommandBufferExecutionRangeMake
+    substituteInPlace PyObjCTest/test_mtlindirectcommandbuffer.py \
+      --replace 'def test_functions(self):' 'def disabled_test_functions(self):'
```
pytestCheckHook.

Mic92 commented on pkgs/development/python-modules/pyobjc-framework-iTunesLibrary/default.nix:
```diff
@@ -0,0 +1,51 @@
+  preCheck = ''
+    # testConstants in PyObjCTest.test_cfsocket.TestSocket returns: Segmentation fault: 11
+    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
+    # Remove Test which is probably missing a sdk check
+    substituteInPlace PyObjCTest/test_itlibmediaitem.py \
+      --replace 'def testConstants_missing(self):' 'def disabled_testConstants_missing(self):'
```
pytestCheckHook.

Mic92 reviewed (changes requested):
I had a look at all 121 files. To move this forward, could you move all the changes done to apple sdk frameworks in a seperate PR? I would like to get them reviewed by darwin maintainers. I added a lot of comments regarding moving stuff away from patching to pytestCheckHook under the assumption that pytest works with pyobjc, correct me here if I am wrong.

veprbl commented:
> I don't see that many redundant parts. Patching `darwin.apple_sdk.sdk.version` into `pyobjc_setup.py` could be moved into a setup hook. Maybe also the files should say that they are auto-generated so people are not starting to modify them directly.

Well, just look at the original https://github.com/NixOS/nixpkgs/pull/21829/commits/28e7f556ce22712197e8da38a21f3eee90d053fd
If I want modify that with something non-trivial, I don't need to work with esh templates because everything is expressed in Nix.

SuperSandro2000 committed:
python3Packages.pyobjc-framework-*: Add comment about template

SuperSandro2000 commented:
> Do you think it will be possible to factorize the expressions for the python packages using the power of Nix

Totally possible but my nix is not good enough for that right now. I would have to fight with type errors a lot and with the error messages in nix 2 it is always a mystery what exactly is wrong.

> Maybe also the files should say that they are auto-generated so people are not starting to modify them directly.

Added.

> could you move all the changes done to apple sdk frameworks in a seperate PR?

I could totally do this but without the context of this PR the changes are not required and don't seem plausible. A reviewer would need to trust me that they are exactly required in this way and could not test if this is really the case. Right now it can be easily tested if my changes are correct or not by removing the change again and rebuilding pyobjc which then should throw an error.

> I would like to get them reviewed by darwin maintainers.

We need a darwin maintainer to review this anyway because this entire PR does not build on anything else.

> I added a lot of comments regarding moving stuff away from patching to pytestCheckHook under the assumption that pytest works with pyobjc, correct me here if I am wrong.

This project does not run with pytest. It complains about outdated types a lot and all tests fail. The custom code around ``python setup.py test`` does not allow test skipping.

some example output:
```
PyObjCTest/test_nswindow.py:13
/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-python3.8-pyobjc-framework-Cocoa-6.2.2.drv-0/pyobjc-framework-Cocoa-6.2.2/PyObjCTest/test_nswindow.py:13: PytestCollectionWarning: cannot collect test
class 'TestNSWindowHelper' because it has a __new__ constructor (from: PyObjCTest/test_nswindow.py)
class TestNSWindowHelper(AppKit.NSObject):
-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED PyObjCTest/test_appcategories.py::TestAppKitCategories::testAnimationContext
FAILED PyObjCTest/test_appcategories.py::TestAppKitCategories::testSavedGraphicsState
FAILED PyObjCTest/test_apphelper.py::TestAppHelper::test_callAfter - TypeErro...
FAILED PyObjCTest/test_apphelper.py::TestAppHelper::test_callLater - TypeErro...
FAILED PyObjCTest/test_apphelper.py::TestAppHelper::test_endSheetMethod - Typ...
```

I created https://github.com/ronaldoussoren/pyobjc/issues/330 to ask if tests could be skipped but did not receive an answer yet.

> Well, just look at the original 28e7f55
> If I want modify that with something non-trivial, I don't need to work with esh templates because everything is expressed in Nix.

The idea behind that one is flawed because other python packages can directly import pyobjc or any number of the frameworks which is not possible with his approach and also all tests where skipped.

siraben commented:
PR looks a little stalled. I can take some time to build pyobjc on macOS and see if it fixes some packages that depend on it.

SuperSandro2000 commented:
https://github.com/NixOS/nixpkgs/commit/3456ef6f300d3882890d09f1fd91d39baaf986c4?w=1 already does a lot which I changed here. PR will probably shrink a bit.

SuperSandro2000 committed:
Imsservice plugin init

SuperSandro2000 committed:
apple_sdk.frameworks: Add MultipeerConnectivity

SuperSandro2000 commented:
> PR looks a little stalled.

Fixed the merge conflict.

> I can take some time to build pyobjc on macOS and see if it fixes some packages that depend on it.

That would be great.

SuperSandro2000 committed:
apple-sdk: only try to copy tbd files if they exist

SuperSandro2000 commented:
@thefloweringash How are darwin-stubs supposed to work when I need to add new frameworks. The source file here https://github.com/NixOS/nixpkgs/commit/3456ef6f300d3882890d09f1fd91d39baaf986c4?w=1#diff-c4d8bbc3261c329aad6d37bf66765868a01dcf4b838d8622d16959d8597ebed8R111 does not exist and I am not sure if I handled this correct with my change.

Also this python package does not work correctly with tbd I think and requires the line removed here https://github.com/NixOS/nixpkgs/commit/3456ef6f300d3882890d09f1fd91d39baaf986c4?w=1#diff-c4d8bbc3261c329aad6d37bf66765868a01dcf4b838d8622d16959d8597ebed8L97. Any advice How to fix that correct?

thefloweringash commented:
> @thefloweringash How are darwin-stubs supposed to work when I need to add new frameworks.

The stubs generator only contains the frameworks required by nixpkgs, which does present a problem when adding new frameworks. I've opened https://github.com/NixOS/darwin-stubs/pull/2 to add the remaining frameworks.

> Also this python package does not work correctly with tbd I think and requires the line removed here [3456ef6?w=1#diff-c4d8bbc3261c329aad6d37bf66765868a01dcf4b838d8622d16959d8597ebed8L97](https://github.com/NixOS/nixpkgs/commit/3456ef6f300d3882890d09f1fd91d39baaf986c4?w=1#diff-c4d8bbc3261c329aad6d37bf66765868a01dcf4b838d8622d16959d8597ebed8L97). Any advice How to fix that correct?

I'll have to look into this. Hopefully the package will handle the lack of framework files on disk, since it's required to support Big Sur.

SuperSandro2000 committed:
apple_sdk.frameworks.Quartz: Fix Headers/IKDeviceBrowserView.h:12:9: fatal error: 'ImageCaptureCore/ImageCaptureCore.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-core: Switch to Foundation from CoreFoundation

SuperSandro2000 committed:
apple_sdk.frameworks.SpriteKit: Fix Headers/SKView.h:34:9: fatal error: 'Cocoa/Cocoa.h' file not found

SuperSandro2000 committed:
python3Packages.pyobjc-framework-CryptoTokenKit: Switch to import check

due to running 0 tests

SuperSandro2000 committed:
python3Packages.pyobjc-framework-LaunchServices: Switch to import check

due to running 0 tests

thefloweringash commented on pkgs/os-specific/darwin/apple-sdk/default.nix:
```diff
@@ -106,6 +106,8 @@ let
           cp -R "${sdk.out}/Library/Frameworks/$name.framework/Versions/$current/Headers" .
         fi
 
+        ln -s -L "/System/Library/Frameworks/$nested_path/Versions/$current/$name"
```
This is bringing back the impure symlinks to system frameworks that no longer exist on Big Sur, and in combination with  "apple-sdk: only try to copy tbd files if they exist", this is almost completely reverting Big Sur support.

SuperSandro2000 replied:
After fixing the wrong stat this is no longer necessary.

thefloweringash reviewed (changes requested):
Some of the changes here revert Big Sur support. If that's something you want to propose that should be a separate PR, but I'm assuming that's not intended. Looking at [PyObjC supported platforms][] it claims to run on macOS 10.9 or later, so the 10.12 support in nixpkgs should be sufficient. What kind of problems were you seeing when using the framework stubs? I see a few failures in check phases, but I can't tell if they're related to the framework stubs or not.

[PyObjC supported platforms]: https://pyobjc.readthedocs.io/en/latest/#supported-platforms

SuperSandro2000 commented:
> What kind of problems were you seeing when using the framework stubs?

I did not try to use them.

SuperSandro2000 commented on pkgs/os-specific/darwin/apple-sdk/default.nix:
```diff
@@ -107,8 +107,8 @@ let
-        if [ "${name}" != "Kernel" ]; then
-          # The Kernel.framework has headers but no actual library component.
+        # Skip frameworks which have no generated tbd file.
+        if stat --printf="" "$tbd_source/*.tbd" 2>/dev/null; then
```
> The tbd file should be at $(nix-build -A darwin.apple_sdk.frameworks.Foundation)/Library/Frameworks/Foundation.framework/Foundation.tbd, but it doesn't exist on this branch.

I found the issue about this and fixed it. 


The change is required until https://github.com/NixOS/darwin-stubs/pull/2 is merged or otherwise I'll get
```
copying path '/nix/store/akqlhgsf02vzhw330nrbd7hrzdh11i68-apple-framework-ScriptingBridge' from 'https://cache.nixos.org'...
copying path '/nix/store/akw65s6c62zx5xwi4pcy76jnavgxpm9j-apple-framework-ContactsPersistence' from 'https://cache.nixos.org'...
copying path '/nix/store/mbh30nci66wagn19zsyz9vc1mym3mfw9-apple-framework-JavaScriptCore' from 'https://cache.nixos.org'...
building '/nix/store/7lhnl21i31lmkkp9kgh7sn8bh2pnvmpd-apple-framework-Contacts.drv'...
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
cp: missing destination file operand after '.'
Try 'cp --help' for more information.
builder for '/nix/store/7lhnl21i31lmkkp9kgh7sn8bh2pnvmpd-apple-framework-Contacts.drv' failed with exit code 1
building '/nix/store/v0a0v5ilbn33hxqrly8bgcd5fg9hicb3-apple-framework-CoreVideo.drv'...
cannot build derivation '/nix/store/vz957sq4ka380j1n3sfq9f35bzzq691h-python3.8-pyobjc-framework-Contacts-6.2.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/89wpq0spvff4qvk6f3hpslb9lbnfdw9s-python3.8-pyobjc-framework-ContactsUI-6.2.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6vwlfy26xxjn5sfnazw67262h99f3pww-python3.8-pyobjc-6.2.2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/6vwlfy26xxjn5sfnazw67262h99f3pww-python3.8-pyobjc-6.2.2.drv' failed
```

SuperSandro2000 committed:
python3Packages.pyobjc: Fix objc4 detection

SuperSandro2000 committed:
python3Packages.pyobkc-frameworks-*: Fix issues when sandbox is enabled

SuperSandro2000 committed:
python3Packages.pyobjc: 6.2 -> 6.2.2, cleanup packages that work with
the current sdk version

SuperSandro2000 committed:
python3Packages.pyobjc: Allow specificing version in update script, fix
typos, cleanups

SuperSandro2000 committed:
apple_sdk.frameworks: Add CloudKit

SuperSandro2000 committed:
apple_sdk.frameworks: Add FinderSync

SuperSandro2000 committed:
apple_sdk.frameworks: Add MediaLibrary

SuperSandro2000 committed:
apple_sdk.frameworks.QTKit: Fix error when python enviroment is created
and /System/Library/Frameworks/QuickTime.framework does not exist

SuperSandro2000 commented:
In the last force push I removed all apple framweworks which get added in the future sdk versions, pyobjc is no longer build with future packages and pyobjc can be build with sandboxing on.

SuperSandro2000 commented:
@ofborg build python3Packages.pyobjc
